### PR TITLE
Serial codec and fixes for some payloads

### PR DESCRIPTION
**Pull request by evandp**

### CODE REVIEW

File: simplebgc/src/commands/read_params.rs [hunk 1]
```diff
@@ -188,32 +180,36 @@ pub enum RcMixChannel {
...
+    pub rc_mix_channel: RcMixChannel,
+}
+
+impl Payload for RcMix {
+    fn from_bytes(b: Bytes) -> Result<Self, PayloadParseError>
+    where
+            Self: Sized {
+        // We expect one byte
+        if b.len() < 1 {
+            return Err(PayloadParseError::InvalidFlags{ name: "RcMixRate".into() });
```
Comment (laptou) [inline simplebgc/src/commands/read_params.rs]: nit: space before `{`

File: simplebgc/src/commands/read_params.rs [hunk 2]
```diff
@@ -188,32 +180,36 @@ pub enum RcMixChannel {
...
+        }
+        let byte = b[0];
+        let rc_mix_rate = 0b00011111 & byte;
+        let rc_mix_channel = RcMixChannel::from_u8(byte >> 5).unwrap();
```
Comment (laptou) [inline simplebgc/src/commands/read_params.rs]: add a comment explaining that bits 0..5 are mix rate and 6..7 are mix channel. you can figure it out by reading the code, but just to reduce cognitive load

File: simplebgc/src/commands/read_params.rs [hunk 3]
```diff
@@ -188,32 +180,36 @@ pub enum RcMixChannel {
...
+        })
     }
 
-    fn from_u64(b: u64) -> Option<Self> {
-        FromPrimitive::from_u8(b as u8)
+    fn to_bytes(&self) -> Bytes
+    where
+            Self: Sized {
+        let mut b = BytesMut::new();
+        let byte = self.rc_mix_rate & (self.rc_mix_channel.to_u8().unwrap() << 5);
```
Comment (laptou) [inline simplebgc/src/commands/read_params.rs]: shouldn't this be `|`, not `&`?

File: simplebgc/src/commands/read_params.rs [hunk 4]
```diff
@@ -51,8 +51,8 @@ payload_rpy!(AxisRcParams, 8);
 #[repr(u8)]
 pub enum AxisRcMode {
     Angle = 1 << 0,
-    Speed = 1 << 1,
-    Inverted = 1 << 2,
+    Fixed = 1 << 1,
+    Inverted = 1 << 3,
 }
```
Comment (laptou) [inline simplebgc/src/commands/read_params.rs]: were you able to determine whether the `Angle` mode corresponds to bits 0..2 = `000` or whether it corresponds to bit 0 = 1?

File: simplebgc/src/commands/read_params.rs (hunk 3 above)
Comment (evandp) [reply simplebgc/src/commands/read_params.rs]: Yep you're right.

File: simplebgc/src/commands/read_params.rs (hunk 1 above)
Comment (evandp) [reply simplebgc/src/commands/read_params.rs]: Fixed

File: simplebgc/src/commands/read_params.rs (hunk 2 above)
Comment (evandp) [reply simplebgc/src/commands/read_params.rs]: Done

File: simplebgc/src/commands/read_params.rs (hunk 4 above)
Comment (evandp) [reply simplebgc/src/commands/read_params.rs]: So the value of the first three bits corresponds to whether it's `AngleRegular`, `AngleTracking` or `Speed`. The fourth bit is a flag for whether the mode is inverted or not.

File: simplebgc/src/commands/read_params.rs [hunk 5]
```diff
@@ -47,12 +47,15 @@ pub struct AxisRcParams {
...
 pub enum AxisRcMode {
-    Angle = 1 << 0,
-    Speed = 1 << 1,
-    Inverted = 1 << 2,
+    AngleRegular = 0,
+    Speed = 1,
+    AngleTracking = 2,
+    InvertedAngleRegular = 8,
+    InvertedSpeed = 9,
+    InvertedAngleTracking = 10,
```
Comment (laptou) [inline simplebgc/src/commands/read_params.rs]: Consider specifying these numbers as binary literals (`0b0000`, `0b1000`) so it's clearer how they are related.

**Review by laptou — APPROVED** [review]
LGTM

File: simplebgc/src/commands/read_params.rs (hunk 5 above)
Comment (evandp) [reply simplebgc/src/commands/read_params.rs]: done

**Pull request merged**